src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/EditProjectFileCommand.cs

```diff
@@ -60,31 +74,36 @@ internal abstract class AbstractEditProjectFileCommand : AbstractSingleNodeProje
         {
             if (!ShouldHandle(node)) return false;
 
-            var projectFileName = Path.GetFileName(_unconfiguredProject.FullPath);
-
-            var projPath = await GetFileAsync(projectFileName).ConfigureAwait(false);
             await _threadingService.SwitchToUIThread();
-            IVsWindowFrame frame;
 
-            frame = _shellUtilities.OpenDocumentWithSpecificEditor(_serviceProvider, projPath, XmlEditorFactoryGuid, Guid.Empty);
+            // If the window has already been opened and hasn't yet been closed, reshow it.
+            if (_isInitialized)
+            {
+                _frame.Show();
+                return true;
+            }
+
+            await SetupFileAsync().ConfigureAwait(true);
 
             IMsBuildModelWatcher watcher = _watcherFactory.CreateExport();
-            await watcher.InitializeAsync(projPath).ConfigureAwait(true);
+            await watcher.InitializeAsync(_tempProjectPath, _lastWrittenXml).ConfigureAwait(true);
+
+            _frame = _shellUtilities.OpenDocumentWithSpecificEditor(_serviceProvider, _tempProjectPath, XmlEditorFactoryGuid, Guid.Empty);
```

Comment:
I'm seeing some odd failures when I use OpenDocument instead of specifying the editor. I'll file a bug for RC3, but I'm going to revert back to specifying the editor for RC2.